trustfall_core/src/schema/adapter/tests.rs

```diff
@@ -99,6 +99,55 @@ fn check_vertex_type_properties_using_one_of() {
     assert_eq!(expected_rows.as_slice(), rows);
 }
 
+#[test]
+fn check_schema_then_vertex_type() {
+    let query = r#"
+{
+    Schema {
+        vertex_type {
+            name @filter(op: "one_of", value: ["$name"]) @output
+
+            property {
+                property: name @output
+            }
+        }
+    }
+}"#;
+    let args = btreemap! {
+        "name".into() => vec!["VertexType", "Property"].into(),
+    }
+    .into();
+    let adapter = Arc::new(SchemaAdapter::new(&SCHEMA));
+
+    let indexed = crate::frontend::parse(&SCHEMA, query).expect("not a valid query");
+    let mut rows: Vec<_> = crate::interpreter::execution::interpret_ir(adapter, indexed, args)
+        .expect("execution error")
+        .collect();
+
+    rows.sort_by(|a, b| a["name"].partial_cmp(&b["name"]).expect("to be comparable"));
```

Comment:
this is actually a problem in the test above aswell, we compared based on "property" rather than "name" so the sort is unstable, it needs to be fixed there too in a future pr

Reply:
Could you fix it here by using the `TryIntoStruct` trait into a struct with a `#[derive(PartialOrd, Ord)]` which will then provide a guaranteed sort order?

We should apply the same fix in the other test as well in a subsequent PR.